backend/main/src/Services/InvitationHashService/InvitationHashService.ts

```diff
@@ -0,0 +1,132 @@
+	private nishikiDynamoDBClient: NishikiDynamoDBClient;
+	private groupRepository: IGroupRepository;
+
+	constructor(
+		nishikiDynamoDBClient: NishikiDynamoDBClient,
+		groupRepository: IGroupRepository,
+	) {
+		this.nishikiDynamoDBClient = nishikiDynamoDBClient;
+		this.groupRepository = groupRepository;
+	}
+
+	/**
+	 * This method generate an invitation hash.
+	 * Before generating an invitation hash, this method check existence of the invitation hash.
+	 * If the invitation hash does exist, this method checks the expiry datetime.
+	 * If the expiry datetime is over, this method generate a new invitation hash.
+	 * If not over, this method returns the invitation hash with updating the expiry datetime.
+	 * The invitation hash link is generated by a string of "groupId + expiry Datetime" using MD5.
+	 * https://genesis-tech-tribe.github.io/nishiki-documents/proect-document/specifications/detail/invitation-to-group#invitation-link-specification
+	 * @param input
+	 */
+	async generateAnInvitationHash(input: {
+		groupId: string;
+		userId: string;
+	}): Promise<
+		Result<IHash, DomainObjectError | GroupNotFound | PermissionError>
+	> {
+		const groupIdOrError = GroupId.create(input.groupId);
+		const userIdOrError = UserId.create(input.userId);
+
+		if (groupIdOrError.err || userIdOrError.err) {
+			if (groupIdOrError.err) {
+				return Err(groupIdOrError.error);
+			}
+			return Err(userIdOrError.unwrapError());
+		}
+
+		const { groupId, userId } = {
+			groupId: groupIdOrError.value,
+			userId: userIdOrError.value,
+		};
+
+		const group = await this.groupRepository.find(groupId);
+
+		if (!group) {
+			return Err(new GroupNotFound("Group not found"));
+		}
+
+		if (!group.canEdit(userId)) {
+			return Err(
+				new PermissionError("You don't have permission to access this group."),
+			);
+		}
+
+		const invitationHash = await this.nishikiDynamoDBClient.getInvitationLink(
+			groupId.id,
+		);
+
+		const now = new Date(Date.now());
+		const expiryDatetime = new Date(Date.now());
+		expiryDatetime.setHours(expiryDatetime.getHours() + 24); // 24 hours.
+
+		const hash = Md5(`${groupId.id}${expiryDatetime}`).toString();
+
+		// invitation link doesn't exist.
+		if (!invitationHash) {
+			await this.nishikiDynamoDBClient.addInvitationLink(
+				groupId.id,
+				expiryDatetime,
+				hash,
+			);
+			return Ok({
+				hash,
+				expiryDatetime,
+			});
+		}
+
+		// invitation link doesn't expire.
+		if (now < invitationHash.linkExpiryTime) {
+			await this.nishikiDynamoDBClient.addInvitationLink(
+				groupId.id,
+				expiryDatetime,
+				invitationHash.invitationLinkHash,
+			);
+			return Ok({
+				hash: invitationHash.invitationLinkHash,
+				expiryDatetime,
+			});
+		}
+
+		// invitation link expires, remove it, and generate a new one.
+		await Promise.all([
+			this.nishikiDynamoDBClient.addInvitationLink(
+				groupId.id,
+				expiryDatetime,
+				invitationHash.invitationLinkHash,
```

Comment:
I think in this case the hash should be the generated one `hash` instead of existing value `invitationHash.invitationLinkHash`.
If my understanding wrong, please mention it. I'll approve this PR.
```suggestion
				hash,
```

Reply:
Thank you for pointing it out.
That's true.